go/src/koding/klient/uploader/uploader.go

```diff
@@ -0,0 +1,275 @@
+package uploader
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	"koding/kites/keygen"
+	"koding/klient/storage"
+	"koding/logrotate"
+
+	"github.com/boltdb/bolt"
+	"github.com/koding/kite"
+	"github.com/koding/logging"
+)
+
+var defaultLog = logging.NewCustom("uploader", false)
+
+// Options represents arguments required to create a Uploader value.
+type Options struct {
+	KeygenURL string      // required
+	Kite      *kite.Kite  // required
+	Bucket    string      // required
+	Region    string      // required
+	DB        *bolt.DB    // optional; in-memory store if nil
+	Log       kite.Logger // optional; defaultLog if nil
+}
+
+// Logger is a wrapper for kite.Logger that implements logging.Logger.
```

Comment:
The code you linked creates kite.Logger from logging.Logger. Here's the opposite - creating logging.Logger from kite.Logger.